### DISCUSSION TITLE
Is this to hard for beginner ?

### DISCUSSION THREAD

**Discussion by leg3nddk** [SebLague/Chess-Challenge#201] [Q&A]
Just wanna ask, is this to hard for one that just started learning C# and want to join in ?

**Comment by goosehub** [comment]
You'll actually make a very formidable for a chess beginner bot if you do the very basic step of having it check if a move wins material (`IsCapture`), potentially hangs a piece (`SquareIsAttackedByOpponent`), or comes with check (`isCheck`) a little things like that. Learning how to recursively check depth will be key to making something formidable against a good chess opponent.

**Comment by DaveAxiom** [comment]
I would say that in order to work on and submit a bot you would need to:

- Know basic C# -- or Java because it's nearly the same
- Know the Minimax algorithm
- Have an understanding of recursion
- Know Chess

The scope of taking this on would be working within the MyBot::Think() function and getting a good grip on what your work is doing when it is run. The list of things to know and do would be longer for an intermediate programmer.

**Reply by BronsonCarder** [reply]
I have the first one. lol
I feel like I'm doing ok. Just have to accept that I'm not going to be competing against the people who have done this forever. lol

**Reply by carsonetb** [reply]
Of course you should know the minimax algorithm and have an understanding of recursion, but this might be a great way to learn about them, you don't need to have a large understanding of them before starting. Even learning C#, if you've ever programmed in another language before, like me, you should be able to get the hang of it.

**Comment by Moonwalker316** [comment]
IMO this is a great begginner project, if you do well enough you'll be feautured in a video!
I would say you need a basic understanding of C# but not much.
I'd also reccomend learning recursion and Minimax algorithim, and alpha beta pruning.

Alternatively you could try your best to make a bot that is intresting, having a set token limit or other limitations to make a cool bot.
Personally I went for the latter because it seems more intresting, and I made a bot that can only look at the current position.

**Comment by dudecon** [comment]
No, it is not too hard. If you are looking for excuses, you will always find them. If you are looking for permission, you already have it. If you are looking for encouragement, you can do it!

**Comment by BronsonCarder** [comment]
This is my first coding challenge. I suck at chess, and I understand recursion, but not enough to implement it. lol
Make what you can. My bot isn't the absolute best by any means...

But, I built a bot that I can't win against in chess, and that feels really awesome!
You got this!

**Comment by leg3nddk** [comment]
Thank you for the replyes. I have begun trying to understand it all. And taking a look at others work. And sorta learned a bit. But will be funny to see how much i can learn until October :)

**Comment by leg3nddk** [comment]
Well i have now finally made a bot that mostly do the same as evil bot. Just in a really ineffecience way. But learned something now :)

**Comment by jongdetim** [comment]
yes, this is very hard for beginners if you're trying to make a competitive bot. The algorithm will be very confusing, hard to benchmark and hard to debug. Furthermore, doing low-level optimisations to reduce the amount of tokens is very challenging. Of course, if you're just trying to get a working bot that can beat the shipped 'evil bot', it's fine as a beginner project.

**Comment by Austin-Nicholas** [comment]
I'd say go for it, I had something (very bad) running in just a few minutes. honestly if you can get a bot that compiles and knows how to move the pieces, I'd submit it and see how it goes.
